sql/clients_last_seen_v1.sql

```diff
@@ -19,9 +19,26 @@ WITH current_sample AS (
 SELECT
   @submission_date AS submission_date,
   CURRENT_DATETIME() AS generated_time,
+
+  -- This last_date struct is used to record the last date that a particular
+  -- client met various criteria; we record a null date if the client does
+  -- not meet a given criterion.
+  STRUCT (
+    IF(
+      country IN ('US', 'FR', 'DE', 'UK', 'CA'),
+      current_sample.submission_date,
+      previous.last_date.seen_in_tier1_country
+    ) AS seen_in_tier1_country,
+    IF(
+      scalar_parent_browser_engagement_total_uri_count_sum >= 5
+      current_sample.submission_date,
+      previous.last_date.visited_5_uri
+    ) AS visited_5_uri
+  ) AS last_date,
```

Comment:
I'm a little unsure of how this is going to be used, could you provide an example?

Reply:
if i understand correctly, it's for this kind of thing:
```sql
SELECT
  COUNTIF(last_date.visited_5_uri IS NOT NULL) AS active_mau,
  COUNTIF(last_date.visited_5_uri = submission_date) AS active_dau
FROM
  clients_last_seen
```